File: go/abalone/segment.go
package abalone

type Segment struct {
	base        Hex
	length      int
	player      Player
	orientation Direction
}

func (s *Segment) segPieces() []Hex {
	result := make([]Hex, s.length)
	p := s.base
	for i := 0; i < s.length; i++ {
		result[i] = p
		p = p.adjacent(s.orientation)
	}
	return result
}

File: fe2/abalone-js/src/engine/segment.ts
/// <reference path="hex.ts" />
/// <reference path="player.ts" />
/// <reference path="direction.ts" />

module Abalone {
export module Engine {
	export interface Segment {
		basePos: Hex;
		orientation: Direction;
		segLength: number;
		player: Player;
	}


	export function segPieces(s: Segment): Hex[] {
		if (s == null) return [];
		var front = s.basePos;
		var pieces = [front];
		for (var i=0; i<s.segLength-1; i++) {
			front = adjacent(front, s.orientation);
			pieces.push(front);
		}
		return pieces;
	}
}
}